src/DotVVM.Framework/Controls/TemplateHost.cs

```diff
@@ -18,13 +20,38 @@ public class TemplateHost : DotvvmControl
         /// Gets or sets the template that will be rendered inside this control.
         /// </summary>
         [MarkupOptions(AllowBinding = false, MappingMode = MappingMode.Attribute, Required = true)]
-        public ITemplate? ContentTemplate { get; set; }
+        public ITemplate Template
+        {
+            get { return (ITemplate)GetValue(TemplateProperty); }
+            set { SetValue(TemplateProperty, value); }
+        }
+        public static readonly DotvvmProperty TemplateProperty
+            = DotvvmProperty.Register<ITemplate, TemplateHost>(c => c.Template, null);
+
 
 
         protected internal override void OnLoad(IDotvvmRequestContext context)
         {
-            ContentTemplate?.BuildContent(context, this);
+            var placeHolder = new PlaceHolder();
+            Template.BuildContent(context, placeHolder);
```

Comment:
Nullable types should complain somewhere here, please add a simple nullcheck to avoid NRE when users forgets to set the property.
```suggestion
            Template.NotNull("TemplateHost.Template is required").BuildContent(context, placeHolder);
```


Also, I think it would be convenient to add a constructor overload with template as parameter. Maybe we could also add overload to the AppendChildren(ITemplate) helper method that would automatically create this control.